src/askem_beaker/contexts/mira/procedures/python3/visualize_mira_models.py

```diff
@@ -0,0 +1,26 @@
+import base64
+
+from mira.metamodel import TemplateModel
+from mira.modeling.viz import GraphicalModel
+
+
+def __visualize_model(name: str, model: TemplateModel) -> str:
+    graphical_model = GraphicalModel.from_template_model(model)
+    filename = f"_preview_{name}.png"
+    graphical_model.write(filename, format="png")
+    with open(filename, "rb") as f:
+        data = f.read()
+        enc = base64.b64encode(bytes(data))
+        return enc.decode("utf-8")
+
+
+__state = locals()
+__models = {}
+
+for model in {{model_vars}}:
+    model: str
+    __models[model] = {"text/plain": repr(__state[model]), "image/png": __visualize_model(model, __state[model])}
+
+del __state
```

Comment:
By using jinja templating, I think we can clean this up to something like the following.
(Note: I haven't really tested this, so don't take this as a solved solution.)
As you can see, instead of generating and reading from a locals() dict, we are just generating code that access the variables directly.

```suggestion
__models = {
{% for model_var in model_vars %}
    "{{model_var}}":  {"text/plain": repr({{model_var}}), "image/png": __visualize_model("{{model_var}}", {{model_var}})}
{% endfor %}
}
```